### PR TITLE
acora: Compile with newer cython to support python3.7

### PR DESCRIPTION
**Pull request by SurenNihalani**

### CODE REVIEW

File: acora/_acora.py [hunk 1]
```diff
@@ -42,8 +42,8 @@ class _MachineState(object):
     """Trie node state for the automaton.
     """
     def __init__(self):
-        self.id = None
-        self.letter = None
+        self.id = 0
+        self.letter = 0
```
Comment (scoder) [inline acora/_acora.py]: Is this change related?

**Review by scoder — COMMENTED** [review]
Thanks! If you can wait just a little longer, I could base the next release on Cython 0.29, which has some more fixes for Py3.7. It's currently being released. Would be nice if you could grab the latest Cython release candidate from github and try it to make sure it works or if your PR needs any more changes.
https://github.com/cython/cython/archive/0.29rc1.zip

File: acora/_acora.py (hunk 1 above)
Comment (SurenNihalani) [reply acora/_acora.py]: yes. Otherwise Cython complains that unsigned long cannot be assigned None

**Comment by scoder** [comment]
Thanks. There were a couple of more changes needed. v2.2 is released.
